src/main/java/frc/robot/commands/TwoBallAutoCurved.java

```diff
@@ -0,0 +1,81 @@
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.RamseteController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.math.trajectory.TrajectoryConfig;
+import edu.wpi.first.math.trajectory.TrajectoryGenerator;
+import edu.wpi.first.math.trajectory.constraint.DifferentialDriveVoltageConstraint;
+import edu.wpi.first.wpilibj2.command.ParallelDeadlineGroup;
+import edu.wpi.first.wpilibj2.command.RamseteCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Constants;
+import frc.robot.FieldInformation;
+import frc.robot.subsystems.DriveTrain;
+import frc.robot.subsystems.Intake;
+import frc.robot.subsystems.Shooter;
+import frc.robot.subsystems.Vision;
+
+public class TwoBallAutoCurved extends SequentialCommandGroup implements AutoCommandInterface {
+    Trajectory m_trajectory;
+    public TwoBallAutoCurved(Shooter shooter, Intake intake, DriveTrain driveTrain, Vision vision) {
+        var autoVoltageConstraint =
+        new DifferentialDriveVoltageConstraint(
+            new SimpleMotorFeedforward(Constants.ksVolts,
+                                       Constants.kvVoltSecondsPerMeter,
+                                       Constants.kaVoltSecondsSquaredPerMeter),
+            Constants.kDriveKinematics,
+            10);
+
+        TrajectoryConfig config =
+            new TrajectoryConfig(Constants.kMaxSpeed,
+                                Constants.kMaxAcceleration)
+                .setKinematics(Constants.kDriveKinematics)
+                .addConstraint(autoVoltageConstraint)
+                .setReversed(true);
+
+        m_trajectory = TrajectoryGenerator.generateTrajectory(
+            getInitialPose(), 
+            List.of(),
+            FieldInformation.middleBlueBall,
+            config
+        ); 
+
+        RamseteCommand ramseteCommand = new RamseteCommand(
+            m_trajectory,
+            driveTrain::getPose,
+            new RamseteController(Constants.kRamseteB, Constants.kRamseteZeta),
+            new SimpleMotorFeedforward(Constants.ksVolts,
+                                        Constants.kvVoltSecondsPerMeter,
+                                        Constants.kaVoltSecondsSquaredPerMeter),
+            Constants.kDriveKinematics,
+            driveTrain::getWheelSpeeds,
+            new PIDController(Constants.kPDriveVel, 0, 0),
+            new PIDController(Constants.kPDriveVel, 0, 0),
+            driveTrain::tankDriveVolts,
+            driveTrain
+        );
+
+        addCommands(
+            new ParallelDeadlineGroup(
+                ramseteCommand.andThen(() -> driveTrain.tankDriveVolts(0, 0)),
+                new IntakeCommand(intake, Constants.INTAKE_SPEED)
+            ),
+            new TurnTowardsHub(driveTrain, vision).withTimeout(Constants.TURN_TIMEOUT_SECS),
```

Comment:
The camera will be oriented with a large vertical FOV width and a narrow horizontal FOX width.
There is no guarantee that the robot will be able to see the target when it picks up the middle ball.
We should include a ramsete to turn back toward the hub before switching to Vision.

Reply:
John, if you run the current auto in simulation, the robot ends pointing pretty accurately to the hub, so I think until we test, this is OK.